### DISCUSSION TITLE
Roles

### DISCUSSION THREAD

**Discussion by amsalama** [dcm4che/dcm4chee-arc-light#2913] [Q&A] [answered]
Hello,
i have multi-client that push dicom to my dcm4chee-arc v5.
but i need for same client user can post and other user not.
for example client1 : user1 can send dicom, other query.
for query : query only studies that client send only not all studies.

**Comment by vrindanayak** [comment, marked as answer]
@amsalama : By default dockerized secured archive comes pre-configured with 2 roles / users for archive UI. 
Roles : admin / user
Users : admin / user

User `admin` has both roles mapped to it whereas user `user` has only `user` role mapped to it. The archive is also pre-configured with a set of [UI permissions](https://github.com/dcm4che/dcm4chee-arc-light/wiki/UI-Permission-Configuration) - some permissions available for both users whereas some are exclusive only to the `admin` user. 

Depending upon your use case for eg. `Action - Studies - Upload File in study` UI permission has only `admin` role configured to it. This means only `admin` user can upload bulkdata file to a study. You may choose to assign `admin` role to your users which shall have the permission to execute these functions.

For your last point
_for query : query only studies that client send only not all studies._
See [Access Control](https://github.com/dcm4che/dcm4chee-arc-light/wiki/Access-control-based-on-Archive-AEs)

**Comment by gunterze** [comment]
Guess, you does not mean "roles" of users authenticated by Open ID in HTTP requests to the Archive Web UI or RESTful service.
You may configure different sets of [Transfer Capabilities](https://dcm4chee-arc-cs.readthedocs.io/en/latest/networking/config/transferCapability.html) provided by different Archive [Network Application Entities](https://dcm4chee-arc-cs.readthedocs.io/en/latest/networking/config/networkAE.html) which specified [Accepted Calling AE Title(s)](https://dcm4chee-arc-cs.readthedocs.io/en/latest/networking/config/dcmNetworkAE.html#dcmacceptedcallingaetitle).
S.o. [DICOM PS 3.15, H.1.1 Data Model Component Objects](http://dicom.nema.org/medical/dicom/current/output/html/part15.html#sect_H.1.1)

Currently, there is no corresponding feature like "Accepted User Roles" for restricting access by RESTful services associated with a particular Archive AE.

**Comment by amsalama** [comment]
Thanks for Reply : 
but You refer me to UI permission only
Also About API and TCP permission .
for example : Client1 Push to AET1 using API(curl)  or TCP (storescu/tls)
    client1 have 2-users,first  using API for pushing and and second user used  for full permissions 
    also first user can not list devices, can not list AET and so on,but second user can do any thing.
    and also for TCP : first user can storescu but can not query , second user full permissions.

Also Client2 (2-user) Push AET2 only not able to list all DEVICES not able to list AET'S 
but only able to push to AET2,also first user for Client2 can only store to AET2 only and can not query for AET2 

Thanks

I hope very quickly response.

**Comment by gunterze** [comment]
> You may configure different sets of Transfer Capabilities provided by different Archive Network Application Entities which specified Accepted Calling AE Title(s).

refers to Storage, Query/Retrieve by DIMSE Services C-STORE, C-FIND/C-MOVE/C-GET.

**Reply by amsalama** [reply]
Thanks,
" Currently, there is no corresponding feature like "Accepted User Roles" for restricting access by RESTful services associated with a particular Archive AE."

this feature can do in which version ?
Is this going to be done?

**Comment by gunterze** [comment]
> Currently, there is no corresponding feature like "Accepted User Roles" for restricting access by RESTful services associated with a particular Archive AE.

would only applies to RESTful services: STOW/QIDO/WADO.

What are you currently using for storage/query/retrieve?

**Reply by amsalama** [reply]
for restful i create shell script using curl 
for storage i am use dcm4che storescu and other command line from dcm4che.

question : if any clients have token can list and download studies that not related to him and this is critical issue.

**Comment by gunterze** [comment]
So storage can already be restricted by setting Accepted Calling AE Title(s) and removing STOW-RS  from [Web Service Class(s)](https://dcm4chee-arc-cs.readthedocs.io/en/latest/networking/config/webApplication.html).

Will think about how to provide user specific access restriction by RESTful services.  Configuration of "Accepted User Roles" is just one option ...

**Reply by amsalama** [reply]
Can I remove CONFIG-RS from restful ?

**Comment by gunterze** [comment]
Think currently only configured STOW-RS/QIDO-RS/WADO-RS is checked by RESTful services, others only by the Archive UI

**Reply by vrindanayak** [reply]
> Think currently only configured STOW-RS/QIDO-RS/WADO-RS is checked by RESTful services, others only by the Archive UI

STOW-RS/QIDO-RS/WADO-RS and WADO-URI is checked by RESTful services - Yes

`others only by the Archive UI` : @amsalama : This is configurable by UI Permissions - You may choose to edit roles for certain permissions or if a particular permission is deleted, the corresponding function doesn't show up in the UI.
